packages/jest-changed-files/src/git.ts

```diff
@@ -53,33 +53,41 @@ const adapter: SCMAdapter = {
     if (changedSince) {
       const [committed, staged, unstaged] = await Promise.all([
         findChangedFilesUsingCommand(
-          ['diff', '--name-only', `${changedSince}...HEAD`].concat(
+          ['diff', '--name-only', `${changedSince}...HEAD`, '--'].concat(
             includePaths,
           ),
           cwd,
         ),
         findChangedFilesUsingCommand(
-          ['diff', '--cached', '--name-only'].concat(includePaths),
+          ['diff', '--cached', '--name-only', '--'].concat(includePaths),
           cwd,
         ),
         findChangedFilesUsingCommand(
-          ['ls-files', '--other', '--modified', '--exclude-standard'].concat(
-            includePaths,
-          ),
+          [
+            'ls-files',
+            '--other',
+            '--modified',
+            '--exclude-standard',
+            '--',
```

Comment:
note: `git ls-files` won't fail sot not require `--`, just unifying same style  with `git diff`